wikimedia/wikipedia-ios
# Bug fixes for long press back button

**Phabricator:** N/A

### Notes
* When working on my inspiration week project, I discovered 2 bugs from this PR of mine: https://github.com/wikimedia/wikipedia-ios/pull/3640/. This fixes a few bugs with iOS 14's new long press back button behavior:
1. If no location permissions have been granted, the root VC "Places" was never shown due to an early return in `viewWillAppear`.
2. Search VCs show up in two ways: From the tab bar (as a root VC), or when tapping a search icon in an article. Only the first situation had been handled.
3. I also added a bit of safety code to the root VCs.

I just discovered that iPad doesn't reconstitute the article titles properly when coming from a fresh start; I'll fix that in a separate PR.

### Test Steps
1. Run app. Open an article. Open a search screen. Open an article. Long press back button, ensure it says "Search" in the list.
2. Run app. Ensure it doesn't have location permissions. Switch to Places, go into an article, long press back button. Ensure it says "Places" at the root.

### Screenshots/Videos
<img width="333" alt="Screenshot 17" src="https://user-images.githubusercontent.com/9295855/119208955-2923cd80-ba59-11eb-8c74-8ddef7d58ed3.png">
